### PR TITLE
Bug 1464484 - Add landfill sampler to mozetl

### PR DESCRIPTION
**Pull request by acmiyaguchi**
[bug link](https://bugzilla.mozilla.org/show_bug.cgi?id=1464484)

This PR formalizes the landfill-sampler as a mozetl job. This job needs to be submitted via Databricks to utilize a specific IAM role due to inclusion of sensitive data. This job filters out all PII so that it can be used CI jobs.

### CODE REVIEW

**Comment by acmiyaguchi** [comment]
@wlach I've finally got around to adding tests for this job, after finally adding support for Databricks in Airflow. 

It looks like there were some caching issues with tox and CircleCI causing one class of CI failures. It looks like there are some tests in `taar` that are also causing some issues. I'm going to debug a bit further.

**Comment by wlach** [comment]
@acmiyaguchi this looks pretty sane. can you rebase against master? I believe that should fix the taar issues (addressed with b081739f6649811a256c36d585497a2a4fefd494 and dcf100e370321c692cd57b0b4a91fecd0e73ef58). I'll do a last pass-over for nits after that's done.

File: tests/test_landfill_sampler.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    """Update the meta-field using provided snippets."""
+    def _update_meta(snippet):
+        doc = deepcopy(sample_document)
+        doc["meta"].update(snippet)
+        return doc
+
+    def _generate_data(snippets):
+        return spark.sparkContext.parallelize(map(_update_meta, snippets))
+
+    return _generate_data
```
Comment (wlach) [inline tests/test_landfill_sampler.py]: I guess you could just use `lambda` here. Might be a little more concise.

Comment (acmiyaguchi) [reply tests/test_landfill_sampler.py]: I'm going to leave it as-is, I think the named function helps a little bit in documenting what it's doing.

**Comment by acmiyaguchi** [comment]
Thanks @wlach for the review! It'll be nice to finally productionize this job.

**Pull request merged**